exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/aggregate/HashAggTemplate.java

```diff
@@ -193,17 +201,43 @@ public int metricId() {
     }
   }
 
+  // Given hashtable index, figure out the corresponding index in the batch.
+  public int idxInBatch(int parition, int idx) {
+    int totalSize = 0;
+    for (int i=0; i < batchHolders[parition].size(); i++) {
+      if (idx < (totalSize + batchHolders[parition].get(i).batchSize)) {
+        return (idx - totalSize);
+      }
+      totalSize += batchHolders[parition].get(i).batchSize;
+    }
+    throw new IllegalArgumentException("idx must be less than total size");
+  }
+
+  // Given hashtable index, figure out the corresponding batch.
+  public int idxOfBatch(int partition, int idx) {
+    int totalSize = 0;
+    for (int i=0; i < batchHolders[partition].size(); i++) {
+      totalSize += batchHolders[partition].get(i).batchSize;
+      if (idx < totalSize) {
+        return i;
+      }
+    }
+    throw new IllegalArgumentException("idx must be less than total size");
+  }
+
   public class BatchHolder {
 
     private VectorContainer aggrValuesContainer; // container for aggr values (workspace variables)
     private int maxOccupiedIdx = -1;
-    private int batchOutputCount = 0;
+    private int batchSize = 0;
```

Comment:
The word **size** causes some confusion (e.g. size in bytes ?); this is why **count** was originally chosen.